### ISSUE TITLE
Customize generated page title

### ISSUE THREAD

**Issue by ajb** [jekyll/jekyll-archives#63]
❤️ to all the contributors for this great plugin.

It seems like this gem is heavily reliant on its ability to set the page title, which prevents us from achieving our preferred page titles:

![img](http://take.ms/3NJtX)

Is there a workaround? Am I missing something?

**Comment by xHN35RQ** [comment]
Would you like to change the title on all archive pages, or just for specific categories/tags?

**Comment by ajb** [comment]
I'm not really sure yet. Are there implications to each?

On Thu, May 19, 2016 at 10:12 PM, Nick <notifications@github.com> wrote:

> Would you like to change the title on all archive pages, or just for
> specific categories/tags?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/jekyll/jekyll-archives/issues/63#issuecomment-220518843>
>



-- 
Adam Becker
(951) 9-BECKER
@AdamJacobBecker

**Comment by xHN35RQ** [comment]
Maybe. I'm working on a branch trying to add some support for this, as I have a similar need. `jekyll-archives` builds the URL for the archive page from `title` so if you set a custom title for a specific cat/tag it changes the URL. Either we need to build the URL from something other than `title` or we need to set an `alt_title` or `page_title` and provide this data via a `tags.yml` file in `_data`. This is approach I'm working on right now. Any thoughts?

**Comment by xHN35RQ** [comment]
I pushed [a branch](https://github.com/xHN35RQ/jekyll-archives/tree/page-titles) where I'm testing customizable tag and category titles. You can now set custom title prefixes in `_config.yml` using:

```
title_prefix:
tag_title_prefix: 
category_title_prefix: 
```

These strings will be added in front of the generated page title. If no tag or cat specific prefix, the plugin falls back to title_prefix, and if no prefix, just the tag/cat title is used.

You can set a custom title per tag/cat using `_data` files, specifically `tags.yml` and `categories.yml` consisting of the following format:

```
- title: tag title exactly as it appears on the archive page without modification
  custom_title: Modified tag Title, perhaps adding more words, or changing the case
```

**Comment by xHN35RQ** [comment]
@ajb using my branch, you'd set the following in `_config.yml` to achieve the result in your screenshot:

```
tag_title_prefix: 'Posts tagged with'
```

And in your tag archive layout, remove everything except `{{ page.title | smartify }}`

**Comment by xHN35RQ** [comment]
@ajb using my branch, you'd set the following in `_config.yml` to achieve the result in your screenshot:

```
tag_title_prefix: 'Posts tagged with'
```

And in your tag archive layout, remove everything from `<h1></h1>` except `{{ page.title | smartify }}`

**Comment by ulrichsg** [comment]
Any news on this issue? I would also like to see it fixed.

**Comment by aarongustafson** [comment]
This is awesome! Any chance we could get a suffix too?

**Comment by lunaryorn** [comment]
I'd also appreciate the ability to customize the title attribute of the generated pages.  Currently it's hard to use this plugin together with jekyll_seo_tag.

**Comment by mckeever02** [comment]
Agreed. How would I go about setting a meta description for the archive pages if I'm using this in line with jekyll-seo-tag?

**Comment by pathawks** [comment]
Would it be too complicated if we allowed a Liquid template to be set as a title in `_config.yml`? I just don't like the idea of `prefix`

```yaml
archives:
    title: "Posts tagged with {{ page.title }}"
```

**Comment by quicoto** [comment]
No news on this?

Thank you.
